### ISSUE TITLE
Your First Game tutorial issues

### ISSUE THREAD

**Issue by BlenderNPR** [godotengine/godot-docs#941]
I tried Your First Game tutorial from start to end. 
http://docs.godotengine.org/en/latest/learning/step_by_step/your_first_game.html 

- I find it is a bit rushed after midway though. Connecting signals need more verbose explanation.

- Few mistake graphics. 
http://docs.godotengine.org/en/latest/_images/add_script_button.png is wrong. 
Node names suddenly changed. 
http://docs.godotengine.org/en/latest/_images/ui_anchor.png
Anchor is Layout in Godot 3.0 beta 2

- Error warnings
```monitoring = false ```
Firing error warnings
```$Collision.disabled = false ```
Changed to that in newer version of the game, but not updated in tutorial.

- Missing code
```$AnimatedSprite.play()```
Missing that code to animate mob.

- No mention on how to edit gradient for the particles.

Good tutorial but a bit hard to follow.

**Comment by cbscribe** [comment]
Thanks for your feedback. This was written prior to Beta 1 so there have been a few changes. I'll add your comments to the revisions I'm doing.

> Connecting signals need more verbose explanation.

I'm curious what you feel might make it clearer?

> Missing code
$AnimatedSprite.play()
Missing that code to animate mob.

This doesn't need to be in code. Playing should be set to "On" in the Inspector.

**Comment by BlenderNPR** [comment]
> Next, click on the **Player** and connect the **hit** signal to the **game_over** function, which will handle what needs to happen when a game ends.

That sentence never mention where the game_over function is located. Few others like that as well.

> Missing code
    $AnimatedSprite.play()
    Missing that code to animate mob.

> This doesn't need to be in code. Playing should be set to "On" in the Inspector.

Should add **Playing: On** in the tutorial as well. Not available in current version.

> No mention on how to edit gradient for the particles.

A simple mention about this would be a huge help.

**Comment by BlenderNPR** [comment]
**Sprites color enhancement**

Another point for possible improvement is to change the color of the mob and player sprites. On a bright background, player sprites are hard to see. On a darker background, mob sprites are hard to see. I suggest that both the mob and player sprites need to be both brighter or both darker, so that they both fit a single value background.

**Comment by cbscribe** [comment]
Fixed by #956 

As for the sprite art, I'd be happy to convert to something better if it were contributed.

**Comment by altertango** [comment]
This sentence is a bit confusing:
"Next, click on the Player and connect the hit signal."
I just couldn't get it. What does it mean by connect...? Finally, by randomly pressing things I double clicked the hit() signal to find the Connecting signal window which it contained the Method In Node input that you are supposed to edit.

**Comment by cbscribe** [comment]
Thanks for the comment.

Note that by the time you get to that part, you have already connected other signals on both the player and the mob objects (`body_entered` on the Player and `screen_exited` on the Mob). The process of connecting signals was described earlier in this section: http://docs.godotengine.org/en/latest/getting_started/step_by_step/your_first_game.html#preparing-for-collisions

**Comment by majkowskir** [comment]
Hi. For me the problem with the sentence "Next, click on the Player and connect the hit signal." lays not within a simple signal connecting process, as described earlier, but with connecting the signal "hit" between two scenes. I've got to cross check my code with the tutorial once more, but I end up with two scenes (Main and Player) with two, separate "master" nodes in each scene and I can't find a way to connect the "hit" signal in  the Palyere scene with the func game_over in the Main scene. I tried to do that via GUI and with Main.gd -> $Player.connect(“hit”, self, “game_over”). 

I suppose I made a mistake somewhere earlier, but I have to agree with BlenderNPR that tutorial is  a little bit confusing.

**Comment by majkowskir** [comment]
OK, I've found my mistake, although I must say there's a step omitted in this tutorial and that is confusing. 

So, here's the thing: at the very begining you're creating a Player scene. There's a sentence "click the “Add/Create a New Node” button and add an Area2D node to the scene." But one step earlier, while creating the scene, one must choose from 2D , 3D, UI or Custom Scene. Nothing about that in tutorial and it's the source of my mistake. The whole tutorial's about  "how to build a 2D game" so 2D seems like an obvious choice. After that, you can create Area2D node as a child and everything goes smoothly till the "hit" signal. Long story short, there's no the hit signal within Main scene. 

Solution: while creating the Player scene, choose "Custom scene" and the Area2D as the root node. After that its instance in the Main scene has the hit signal and you can connect it to the game_over func.

**Comment by majkowskir** [comment]
Oh, now I see... thank you! Still, confusing as hell. Beside that, it's my second evening with godot and I like it soo much! Just feels... right.
